pkgs/development/web/nodejs/nodejs.nix

```diff
@@ -60,8 +61,8 @@ let
       NIX_CFLAGS_COMPILE = "-D__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__=101300";
     };
 
-    CC_host = "cc";
-    CXX_host = "c++";
+    CC_host = if isCrossBitness then "${stdenv.cc.targetPrefix}cc" else "cc";
+    CXX_host = if isCrossBitness then "${stdenv.cc.targetPrefix}c++" else "c++";
```

Comment:
Are we using `$NIX_CC` instead of `$NIX_CC_FOR_BUILD` in this case? If we don’t need C compiler for build platform, perhaps it makes sense to omit `depsBuildBuild` entirely if `isCrossBitness`?

Reply:
Yes it should be possible to just drop `depsBuildBuild` in case of `isCrossBitness`.

Would it even make sense to always use this magic for cross-compilation and not just when bitness differed? The issue with `isCrossBitness` is that as far as I can see, this is only extraction for the Nix-supported platforms, but as you linked, v8 actually supports only limited build platforms not actually matching bitness.

Reply:
>Would it even make sense to always use this magic for cross-compilation and not just when bitness differed?

Yes, sounds reasonable to me. Always using an emulator instead of cross-compiling should make Node.js build faster, see also #321847.

Reply:
It is now converted to build always with the patch and with wrapper.

I tested it on amd64 and cross build from amd64 to armv7.